node/src/actors/chain_manager/mining.rs

```diff
@@ -365,6 +376,25 @@ impl ChainManager {
                 dr_state.data_request.collateral
             };
 
+            // Check if we have enough collateralizable unspent outputs before starting
+            // retrieval
+            let block_number_limit = self
+                .chain_state
+                .block_number()
+                .saturating_sub(collateral_age);
+            if !check_commit_collateral(
+                collateral_amount,
+                &self.chain_state.own_utxos,
+                own_pkh,
+                &self.chain_state.unspent_outputs_pool,
+                timestamp,
+                // The block number must be lower than this limit
+                block_number_limit,
+            ) {
+                log::debug!("Mining data request: Insufficient collateral, the data request need {} mature wits", Wit::from_nanowits(collateral_amount));
```

Comment:
Also this doesn't show the dr_pointer which may be confusing if there are many data requests.

Reply:
Oh nevermind, too late, it's merged now.